src/apps/components/juttle-viewer.js

```diff
@@ -1,9 +1,54 @@
 import React from 'react';
 
-const JuttleViewer = (props) => {
-    return (
-        <pre className='juttle-source'>{props.bundle.program}</pre>
-    );
+import JuttleEditor from './juttle-editor';
+import { newBundle } from '../actions';
+import { LOCAL_BUNDLE_ID } from '../constants';
+
+class JuttleViewer extends React.Component {
+    constructor(props) {
+        super(props);
+        let isNewBundle = !props.bundle;
+        if (isNewBundle) {
+            props.dispatch(newBundle(LOCAL_BUNDLE_ID, {program: ''}, {}));
+        }
+
+        let query = window.location.search || '';
+        let isLocal = query.match(/\?local/);
+        this.state = {
+            readOnly: !isNewBundle && !isLocal
+        };
```

Comment:
instead of doing this, lets set the runMode to local (via the reducer) and do this:
```javascript
this.state = { readOnly: !this.props.runMode.local }
```